lib/data/model.ts

```diff
@@ -169,6 +169,10 @@ export default class Model extends DenaliObject {
     return new Proxy(this, {
 
       get(model: Model, property: string): any {
+        if (property === 'constructor') {
```

Comment:
Ah, yea, good point about the underlying ORM record instance having that constructor property.

I'm not sure we want to special case constructor alone though - that feels potentially brittle. Pretty soon we'll need special cases for any number of properties that _might_ appear on the underlying ORM record. 

I'm thinking we want the Denali API to win if there are any conflicts, otherwise addons can't truly rely on the model interface.

The comment a little further down in this method indicates another potential hazard here. Many ORMs may have functionality that works with non-persisted properties. The problem is:

```
this.db.create('user', { password, passwordConfirmation })
// password is a DB column, but passwordConfirmation is not
// the ORM has validation support, so will fail to save if passwordConfirmation is different, but it doesn't persist the value
```

The adapter will call `buildRecord({ password, passwordConfirmation })`, and if the underlying ORM allows for non-persistent fields to be set this way (i.e. MemoryAdapter), the underlying record will now have the `.passwordConfirmation` property attached. 

This means that when the user tries to `myUser.passwordConfirmation`, this proxy getter will fire, and Denali will see that `passwordConfirmation` isn't an `attr` for this model, and so (without the double check here at the bottom of the proxy get handler) would return `myUser.passwordConfirmation`, rather than `myUser.record.passwordConfirmation`.

Right now, Denali generally assumes that it should only "pass through" reads and writes to defined `attrs`, but this example indicates that perhaps the model proxy should be more generous in passing through operations, and instead only intercept the operations it needs to. Perhaps something like:

```
get(model, property) {
  if (isAttribute(property)) {
    return adapter.getAttribute(model, property);
  }
  if (isRelationshipMethod(property)) {
    return invokeRelationshipMethod(property);
  }
  if (shouldIntercept(property)) { // handle special cases like constructor
    return model[property];
  }
  return model.record[property];
}
```